servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/RequestResponseContextTest.java

```diff
@@ -0,0 +1,243 @@
+                        for (Buffer chunk : request.payloadBody()) {
+                            writer.write(chunk);
+                        }
+                    }
+                }, strategy -> strategy).adaptor();
+                break;
+            default:
+                throw new IllegalStateException("Unknown api: " + api);
+        }
+        super.service(newService);
+    }
+
+    void transferContext(HttpRequestMetaData request, HttpResponseMetaData response) {
+        try {
+            List<String> requestCtxValue = request.context().get(SERVER_REQUEST_KEY);
+            assertThat(requestCtxValue, is(notNullValue()));
+            response.context().put(SERVER_RESPONSE_KEY, requestCtxValue.get(0) + '-' + RESPONSE_CONTEXT_VALUE);
+        } catch (Throwable t) {
+            asyncError.add(t);
+        }
+    }
+
+    @Test
+    void testAsyncAggregated() throws Exception {
+        setUp(Api.AsyncAggregated);
+        HttpClient client = streamingHttpClient().asClient();
+        HttpRequest request = client.get(SVC_ECHO);
+        request.context().put(CLIENT_REQUEST_KEY, singletonList(REQUEST_CONTEXT_VALUE));
+        HttpResponse response = client.request(request).toFuture().get();
+        assertResponse(response.toStreamingResponse(), HTTP_1_1, OK, 0);
+        assertContext(request, response);
+    }
+
+    @Test
+    void testAsyncStreaming() throws Exception {
+        setUp(Api.AsyncStreaming);
+        StreamingHttpClient client = streamingHttpClient();
+        StreamingHttpRequest request = client.get(SVC_ECHO);
+        request.context().put(CLIENT_REQUEST_KEY, singletonList(REQUEST_CONTEXT_VALUE));
+        StreamingHttpResponse response = makeRequest(request);
+        assertResponse(response, HTTP_1_1, OK, 0);
+        assertContext(request, response);
+    }
+
+    @Test
+    void testBlockingAggregated() throws Exception {
+        setUp(Api.BlockingAggregated);
+        BlockingHttpClient client = streamingHttpClient().asBlockingClient();
+        HttpRequest request = client.get(SVC_ECHO);
+        request.context().put(CLIENT_REQUEST_KEY, singletonList(REQUEST_CONTEXT_VALUE));
+        HttpResponse response = client.request(request);
+        assertResponse(response.toStreamingResponse(), HTTP_1_1, OK, 0);
+        assertContext(request, response);
+    }
+
+    @Test
+    void testBlockingStreaming() throws Exception {
+        setUp(Api.BlockingStreaming);
+        BlockingStreamingHttpClient client = streamingHttpClient().asBlockingStreamingClient();
+        BlockingStreamingHttpRequest request = client.get(SVC_ECHO);
+        request.context().put(CLIENT_REQUEST_KEY, singletonList(REQUEST_CONTEXT_VALUE));
+        BlockingStreamingHttpResponse response = client.request(request);
+        assertResponse(response.toStreamingResponse(), HTTP_1_1, OK, 0);
+        assertContext(request, response);
+    }
+
+    void assertContext(HttpRequestMetaData request, HttpResponseMetaData response) {
```

Comment:
In general I really like that the tests here are very concise. However, I do believe it is crucial to generate a value in each test, possibly a randomized one, to reduce risk of programming error such that instead of the constants for context keys, the future developer reuses the constant to feed the returned value. If that happens, those tests will succeed even if there is a real problem. That's why I think for this particular case the transported value from request to response should be custom each time to eliminate the risk of fooling the test by accident (not assuming malice here :)). The keys and header names are fine remaining constants, just the payload should be randomized. Also no need to concatenate with the response context value. We'd be testing the whole path when we assume the request context value travels all the way back unchanged.